### PR TITLE
[SPARK-30048][SQL] Enable aggregates with interval type values for RelationalGroupedDataset

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

Now the min/max/sum/avg are support for intervals, we should also enable it in RelationalGroupedDataset

### Why are the changes needed?

API consistency improvement

### Does this PR introduce any user-facing change?

yes, Dataset support min/max/sum/avg(mean) on intervals
### How was this patch tested?

add ut

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu @HyukjinKwon @wangyum, thanks for reviewing.

File: sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala [hunk 1]
```diff
@@ -98,10 +98,10 @@ class RelationalGroupedDataset protected[sql](
       // Make sure all specified columns are numeric.
       colNames.map { colName =>
         val namedExpr = df.resolve(colName)
-        if (!namedExpr.dataType.isInstanceOf[NumericType]) {
+        if (!TypeCollection.NumericAndInterval.acceptsType(namedExpr.dataType)) {
```
Comment (maropu) [inline sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala]: Can you update the comment to make it more general one?
https://github.com/apache/spark/blob/c2d513f8e9f5d96dae7990325617013bcc92fe5d/sql/catalyst/src/main/scala/org/apache/spark/sql/types/AbstractDataType.scala#L82

Comment (yaooqinn) [reply sql/core/src/main/scala/org/apache/spark/sql/RelationalGroupedDataset.scala]: thanks for your suggestion. Please check 。

File: sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala [hunk 2]
```diff
@@ -268,8 +268,10 @@ class Dataset[T] private[sql](
       }
   }
 
-  private[sql] def numericColumns: Seq[Expression] = {
-    schema.fields.filter(_.dataType.isInstanceOf[NumericType]).map { n =>
+  private[sql] def numericCalculationSupportedColumns: Seq[Expression] = {
+    schema.fields.filter{ f =>
+      TypeCollection.NumericAndInterval.acceptsType(f.dataType)
+    }.map { n =>
       queryExecution.analyzed.resolveQuoted(n.name, sparkSession.sessionState.analyzer.resolver).get
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala]: nit: we can do
```
queryExecution.analyzed.output.filter { attr =>
  TypeCollection.NumericAndInterval.acceptsType(attr.dataType)
}
```

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
cc: @cloud-fan @dongjoon-hyun

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**